canopy_eddyx_mod.F90

```diff
@@ -0,0 +1,111 @@
+module canopy_eddyx_mod
+
+    implicit none
+
+contains
+
+!:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+    SUBROUTINE CANOPY_EDDYX( HCM, ZK, USTAR, MOL, KZ )
```

Comment:
@zmoon As in my previous response to inputting whole profile for the canopy_wind module, it seems this one needs specific HCM and ZK inputs in a loop somewhere because of the conditions in the code, for example:

       hol = HCM/MOL
        tlc = (HCM/USTAR) * (                        &
            (0.256 * (ZK-(0.75*HCM))/HCM ) +      &
            (0.492*EXP((-0.256*ZK/HCM)/0.492)) )
        sigma = 0.0_rk

        IF ( hol .LT. -0.1 )  THEN  !UNSTABLE
            IF ( ZK/HCM .GT. 1.25 ) THEN !SIGMACAN = Eulerian vertical velocity variance
                sigma = 1.25*USTAR
            END IF
            IF ( ZK/HCM .GE. 0.175  .AND.  ZK/HCM .LE. 1.25 ) THEN
                sigma = USTAR * ( 0.75 + (0.5 * COS((pi/1.06818) *     &
                    (1.25 - (ZK/HCM)))) )
            END IF
            IF ( ZK/HCM .LT. 0.175 )  THEN
                sigma = 0.25*USTAR
            END IF
        END IF

Unless you have a better idea, I leave this in the same format.

Reply:
@zmoon See my latest comment on using elemental subroutine in canopy_wind and the errors that result.  I think it may work better here, but if you agree, lets make a new issue to explore making all modules elemental to allow passing whole profiles when we improve codes after this PR.

Reply:
Sounds fine. We can also fake elemental by putting the loop version in another subroutine and joining the two with an interface block, allowing it to be called both ways.